src/app/profile-edit/profile-edit.page.ts

```diff
@@ -26,57 +27,41 @@ export class ProfileEditPage {
     private loader: LoaderService,
     private toastService: ToastService,
     private navCtrl: NavController,
-    private attachment: AttachmentService
+    private attachment: AttachmentService,
+    private profileService: ProfileService
   ) {}
 
   ionViewWillEnter() {
-    this.getFormJsonAndData();
+    this.loadFormAndData();
   }
 
-  async getFormJsonAndData() {
+  async loadFormAndData() {
     await this.loader.showLoading("Please wait while loading...");
-    const handleError = (message: string) => catchError(err => {
-      this.toastService.presentToast(err?.error?.message || message, 'danger');
-      return of({ status: 'error', result: {} });
+    this.profileService.getFormJsonAndData().subscribe(([formJsonRes, profileFormDataRes]: any) => {
+      if (formJsonRes?.status === 200) {
```

Comment:
what happens in the else case for all these if cases? and since the main logic depends on formJson and formData can't we combine all conditions into a single perhaps?